benchmark/scenarios/write_request_load.py

```diff
@@ -208,151 +77,90 @@ def _get_dataset_node(self, nodes):
-        :return: A `Deferred` that fires when the desired scenario is
+        Executes the setup and starts running the write scenario.
+
+        :return: A Deferred that fires when the desired scenario is
             established (e.g. that a certain load is being applied).
         """
-        # The fist thing we need to do before actually running the scenario
-        # is to update the dataset_id, as we need the information to do the
-        # write requests to generate the load
-        self.dataset_id = dataset.dataset_id
-
-        self.loop.start(interval=1)
+        # List all the nodes registered in the control service
+        d = self.control_service.list_nodes()
+        d.addCallback(self._get_dataset_node)
+        # Once we have the list of nodes, we will create a dataset.
+        # We cannot start the scenario until we have a working dataset, so
+        # `_create_dataset` will work like a setup of the write scenario
+        d.addCallback(self._create_dataset)
+        d.addCallback(self._set_dataset_id)
+        return d
 
-        def reached_target_rate():
-            return self.rate_measurer.rate() >= self.request_rate
+    def make_request(self):
+        if self.dataset_node is None:
+            # Rise exception
+            raise DatasetCreationTimeout
+        return self.control_service.move_dataset(
+            self.dataset_node.uuid,
+            self.dataset_id
+            )
 
-        def handle_timeout(failure):
-            failure.trap(CancelledError)
-            raise WRequestRateNotReached
 
-        # Loop until we reach the expected rate, or we timeout
-        waiting_for_target_rate = loop_until(self.reactor,
-                                             reached_target_rate,
-                                             repeat(1))
-        timeout(self.reactor, waiting_for_target_rate, self.timeout)
-        waiting_for_target_rate.addErrback(handle_timeout)
+@implementer(IScenario, IScenarioSetup)
+class WriteRequestLoadScenario(object):
+    """
+    A scenario that places load on the cluster by performing write
+    requests at a specified rate.
 
-        # Start monitoring the scenario as soon as the target rate is reached.
-        def monitor_scenario_status(result):
-            self.monitor_loop.start(interval=1)
+    :ivar reactor: Reactor to use.
+    :ivar cluster: `BenchmarkCluster` containing the control service.
+    :ivar request_rate: The target number of requests per second.
+    :ivar sample_size: The number of samples to collect when measuring
+        the rate.
+    :ivar timeout: Maximum time in seconds to wait for the requested
+        rate to be reached.
+    """
 
-        waiting_for_target_rate.addCallback(monitor_scenario_status)
+    def __init__(self, reactor, cluster, request_rate=10,
+                 sample_size=DEFAULT_SAMPLE_SIZE, timeout=45):
+        self.setup = WriteRequest(reactor, cluster)
+        self.read_request = self.setup
```

Comment:
There doesn't appear to be a good reason to set these as instance fields. :question: